components/javascript-wrapper.js

```diff
@@ -0,0 +1,157 @@
+/**
+ * This file contacts the common code used by all of the variations of any RDF
+ * node componentin the noflo implementation of the RDF Pipeline.
+ */
+
+var _ = require('underscore');
+
+// TODO: Revisit with James' latest work
+var promiseComponent = require('./promise-component');
+var rpf = require('./rpf');
+
+/**
+ * This module provides a Javascript wrapper for updaters.  It is responsible for setting up the
+ * code to ensure updater callback, and then invoking the RDF pipeline framework to complete building
+ * the component and setting up the vni and state metadata.
+ *
+ * @param nodeDef updater node definition. This can be a simple updater function, or an object that contains 
+ *        updater metadata (e.g., input/output port definitions, description, icons), and an updater.
+ *
+ * @return a promise to create the noflo rdf component
+ */
+module.exports = function(nodeDef) { 
+
+    // Save the caller's node instance 
+    var self = this;
+    
+    // Ensure we have at least one input port and and updater, using a default stub updater if 
+    // there is none specified
+    var updaterArgs = null;
+    if ( _.isUndefined(nodeDef) || 
+       ( ! _.isFunction(nodeDef) && ( _.isEmpty( nodeDef ) || ( _.isEmpty( nodeDef.inPorts ))))) {
+       // Got nothing to work with - no input ports and no updater, so create a default updater and input port 
+       nodeDef = { 
+           inPorts: {
+               input: {
+                   datatype: 'object',
+                   description: "default input port" 
+               }
+           },
+           updater: defaultUpdater
```

Comment:
Suggestion: rename defaultUpdater -> defaultJavascriptUpdater, because there will be other kinds of updaters, and this one is for the JavascriptWrapper.